### PR TITLE
various fixes about variant page

### PR DESCRIPTION
**Pull request by JiaoJiao123**

### CODE REVIEW

File: app/scripts/services/utils.js [hunk 1]
```diff
@@ -52,12 +52,13 @@ angular.module('oncokbStaticApp')
                     'NA') : 'NA';
             },
             getVariantCellContent: function(geneName, variantName) {
-                if (geneName && variantName && variantName.indexOf('/') === -1) {
-                    geneName = geneName.trim();
-                    variantName = variantName.trim();
-                    return '<a href="#/gene/' + geneName + '/variant/' + variantName + '">' + variantName + '</a>';
+                geneName = geneName.trim();
```
Comment (zhx828) [inline app/scripts/services/utils.js]: Keep `if(geneName & variantName)`

File: app/scripts/services/utils.js [hunk 2]
```diff
@@ -52,12 +52,13 @@ angular.module('oncokbStaticApp')
...
+                variantName = variantName.trim();
+                var formattedVariantName = variantName;
+                if (/\//.test(variantName)) {
+                    formattedVariantName = encodeURIComponent(encodeURIComponent(formattedVariantName));
```
Comment (zhx828) [inline app/scripts/services/utils.js]: Why encodeURIComponent twice?

File: app/scripts/services/utils.js [hunk 3]
```diff
@@ -52,12 +52,13 @@ angular.module('oncokbStaticApp')
...
                 }
-                return '<span>' + variantName + '</span>';
+                return '<a href="#/gene/' + geneName + '/variant/' + formattedVariantName + '">' + variantName + '</a>';
```
Comment (zhx828) [inline app/scripts/services/utils.js]: If the geneName and variantName are invalid, return empty string

File: app/scripts/services/utils.js (hunk 2 above)
Comment (JiaoJiao123) [inline app/scripts/services/utils.js]: If you encode it only once, the server decodes it, then it's the same as not encoding it at all.

**Pull request merged**